### PR TITLE
Fixes #35127 - migrate discovery and bootdisk templates to Foreman

### PR DESCRIPTION
**Pull request by nofaralfasi**

### CODE REVIEW

File: app/views/unattended/provisioning_templates/bootdisk/generic_efi_host.erb [hunk 1]
```diff
@@ -0,0 +1,76 @@
+<%#
+name: Boot disk Grub2 EFI - generic host
+snippet: false
+model: ProvisioningTemplate
+kind: Bootdisk
+description: |
+  Boot disk Grub2 EFI - generic host
+-%>
```
Comment (ofedoren) [inline app/views/unattended/provisioning_templates/bootdisk/generic_efi_host.erb:11]: Don't we need to add requirements for bootdisk plugin in order to be able actually use these templates? Something like https://github.com/theforeman/foreman/blob/develop/app/views/unattended/report_templates/host_-_applicable_errata.erb#L20

**Review by ofedoren — COMMENTED** [review]
Just my two cents...

File: app/views/unattended/provisioning_templates/bootdisk/generic_efi_host.erb (hunk 1 above)
Comment (nofaralfasi) [reply app/views/unattended/provisioning_templates/bootdisk/generic_efi_host.erb:11]: You are right, thank you!

**Comment by nofaralfasi** [comment]
Still needs to fix the failing tests on foreman_bootdisk and on foreman_discovery plugins.

**Review by ofedoren — DISMISSED** [review]
I think you can't fix the tests in plugins until this is merged.

ACK from me :)

**Comment by stejskalleos** [comment]
[test katello]

File: app/models/template_kind.rb [hunk 2]
```diff
@@ -25,6 +25,8 @@ def self.default_template_labels
       "cloud-init" => N_("Cloud-init template"),
       "host_init_config" => N_("Host initial configuration template"),
       "registration" => N_("Registration template"),
+      "kexec" => N_("Discovery Kexec template"),
+      "Bootdisk" => N_("Boot disk embedded template"),
```
Comment (stejskalleos) [inline app/models/template_kind.rb]: I think we can name it simply `Discovery Kexec` & `Boot disk`

**Review by stejskalleos — CHANGES_REQUESTED** [review]
I have only one nitpick about naming the kinds, otherwise works well.
@nofaralfasi Please renamed the kinds and if we will have :green_apple: run I'll merge it.

**Comment by Ron-Lavi** [comment]
[test foreman]

**Comment by Ron-Lavi** [comment]
test foreman

**Comment by stejskalleos** [comment]
[test foreman]

File: test/static_fixtures/templates/provision/redhat_kexec.erb [hunk 3]
```diff
@@ -0,0 +1,60 @@
+<%#
```
Comment (stejskalleos) [inline test/static_fixtures/templates/provision/redhat_kexec.erb]: This is for `foreman_discovery` plugin right? If yes than the file should be in the plugin repository, not here

**Review by stejskalleos — CHANGES_REQUESTED** [review]

File: test/static_fixtures/templates/provision/redhat_kexec.erb (hunk 3 above)
Comment (nofaralfasi) [reply test/static_fixtures/templates/provision/redhat_kexec.erb]: I fixed it. Sorry for the misunderstanding.

**Review by stejskalleos — APPROVED** [review]
:green_apple:

**LGTM**, tests are green, let's get this in!
Thanks @nofaralfasi and @ofedoren

**Pull request merged**